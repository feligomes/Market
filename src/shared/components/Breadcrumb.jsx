import React, { useContext } from "react";
import Breadcrumbs from "@mui/material/Breadcrumbs";
import Typography from "@mui/material/Typography";
import Link from "@mui/material/Link";
import { PRODUCT_SCREEN, DASHBOARD_SCREEN } from "../constants/screenNames";
import Context from "../../context/Context";
import NavigateNextIcon from "@mui/icons-material/NavigateNext";

const Breadcrumb = ({ actualScreen }) => {
  const { redirect } = useContext(Context);

  const handleClick = () => {
    redirect(DASHBOARD_SCREEN);
  };

  const breadcrumbs = (actualScreen) => [
    <Link
      href="#"
      underline="none"
      key="1"
      color="inherit"
      onClick={handleClick}
    >
      Home
    </Link>,
    <Typography key="2" color="text.primary">
      {actualScreen === PRODUCT_SCREEN ? "Detail" : "Cart"}
    </Typography>,
  ];

  return (
    <Breadcrumbs
      separator={<NavigateNextIcon fontSize="small" />}
      aria-label="breadcrumb"
      sx={{
        marginBottom: "20px",
      }}
    >
      {breadcrumbs(actualScreen)}
    </Breadcrumbs>
  );
};

export default Breadcrumb;
