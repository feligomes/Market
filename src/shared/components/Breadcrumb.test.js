import React from "react";
import renderer from "react-test-renderer";
import { Provider } from "../../context/Context";
import Breadcrumb from "./Breadcrumb";

it("renders correctly", () => {
  const mockRedirect = (value) => value;
  const mockActualPage = (value) => value;
  const tree = renderer
    .create(
      <Provider value={{ actualPage: mockActualPage, redirect: mockRedirect }}>
        <Breadcrumb />
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
