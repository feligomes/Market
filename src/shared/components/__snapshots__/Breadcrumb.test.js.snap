// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<nav
  aria-label="breadcrumb"
  className="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-1a0xqkg-MuiTypography-root-MuiBreadcrumbs-root"
>
  <ol
    className="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
  >
    <li
      className="MuiBreadcrumbs-li"
    >
      <a
        className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1n8yj6x-MuiTypography-root-MuiLink-root"
        href="#"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
      >
        Home
      </a>
    </li>
    <li
      aria-hidden={true}
      className="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
        data-testid="NavigateNextIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
        />
      </svg>
    </li>
    <li
      className="MuiBreadcrumbs-li"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-1gqyz35-MuiTypography-root"
      >
        Cart
      </p>
    </li>
  </ol>
</nav>
`;
